lib/lexer_mdx.mll

```diff
@@ -64,12 +64,12 @@ and cram_text section = parse
       { let section = (String.length n, str) in
         newline lexbuf;
         `Section section :: cram_text (Some section) lexbuf }
-  | "  " ([^'\n']* as first_line) eol
+  | ("  " as ws) ([^'\n']* as first_line) eol
       { let start = Lexing.lexeme_start_p lexbuf in
         newline lexbuf;
         let header = "sh" in
         let requires_empty_line, contents = cram_block lexbuf in
-        let contents = first_line :: contents in
+        let contents = (Format.asprintf "%s%s" ws first_line) :: contents in
```

Comment:
This is because by matching on `ws` in the lexer rule it eats the whitespace, so `first_line` doesn't contain the leading whitespace. An important aspect of this refactoring would be to preserve the whitespace, so I added the whitespace back on.

A problematic aspect is that due to the lexer in MDX files the `cram` blocks have to be indented by exactly two spaces, so indenting by four wouldn't be accepted but I deemed this an aspect that can be improved at a later point in time, the rest of the code should be able to work with any amount of `hpad` as long as it is consistent.